xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecPassthrough.cpp

```diff
@@ -7,13 +7,18 @@
  */
 
 #include "DVDAudioCodecPassthrough.h"
+
 #include "DVDCodecs/DVDCodecs.h"
 #include "DVDStreamInfo.h"
+#include "ServiceBroker.h"
+#include "settings/Settings.h"
+#include "settings/SettingsComponent.h"
```

Comment:
This is not good. We now have a direct dependency of a low level components to Settings, SettingsBroker. Means this Packer here gains knowledge of stuff it should not need to have at all, e.g. wrong direction of knowledge.

The upper layer rather should tell it on construction what it has to do.

Reply:
Moved to VideoPlayerAudio.cpp on codec creation

Reply:
Hehe, yeah - seen that, though I think we should move this information into CProcessInfo, which is the place that can be overwritten by platform, e.g. the Android platform. That way we would use proper object orentation and can keep the settings also out of AudioCodec. On ProcessInfo a default method e.g. WantsRawPassthrough() which is false by default could be overwritten by Android and answered depending on the settings there by override.

What do you think?

Reply:
Seems is better and more easy 😮